lib/buffer.js

```diff
@@ -264,7 +264,8 @@ function fromObject(obj) {
   }
 
   if (obj) {
-    if (isArrayBuffer(obj.buffer) || 'length' in obj) {
+    if (isArrayBuffer(obj.buffer) || 'length' in obj ||
+        isSharedArrayBuffer(obj)) {
```

Comment:
while we're at it can we do something about `'length' in obj`? by "do something" i mean let's just simplify it for `!!obj.length` or some such. using `in` here is wasting a lot of performance.